core-bundle/src/Cache/EntityCacheTags.php

```diff
@@ -0,0 +1,319 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cache;
+
+use Contao\Model;
+use Contao\Model\Collection as ModelCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\Persistence\Mapping\MappingException;
+use FOS\HttpCache\CacheInvalidator;
+use FOS\HttpCache\ResponseTagger;
+
+/**
+ * Use this helper service to derive 'contao.db.*' cache tags from entity/model
+ * classes and instances. The tagWith*() and invalidateFor*() shortcut methods
+ * directly tag the response or invalidate tags. If your application does not
+ * use response tagging, these methods are no-ops.
+ */
+class EntityCacheTags
+{
+    private EntityManagerInterface $entityManager;
+    private ?ResponseTagger $responseTagger;
+    private ?CacheInvalidator $cacheInvalidator;
+
+    /**
+     * @var array<string, ClassMetadata>
+     */
+    private array $classMetadata = [];
+
+    public function __construct(EntityManagerInterface $entityManager, ResponseTagger $responseTagger = null, CacheInvalidator $cacheInvalidator = null)
+    {
+        $this->entityManager = $entityManager;
+        $this->responseTagger = $responseTagger;
+        $this->cacheInvalidator = $cacheInvalidator;
+    }
+
+    /**
+     * Derives a cache tag from an entity class and returns it.
+     */
```

Comment:
```suggestion
     *
     * @param class-string $className
     */
```
Should we annotate the type `class-string` here so that the IDE and our analysis tools know this should not be some other string?